src/operations/drop.ts

```diff
@@ -24,10 +31,81 @@ export class DropCollectionOperation extends CommandOperation<boolean> {
     session: ClientSession | undefined,
     callback: Callback<boolean>
   ): void {
-    super.executeCommand(server, session, { drop: this.name }, (err, result) => {
-      if (err) return callback(err);
-      if (result.ok) return callback(undefined, true);
-      callback(undefined, false);
+    (async () => {
+      const db = this.db;
+      const options = this.options;
+      const name = this.name;
+
+      const encryptedFieldsMap = db.s.client.options.autoEncryption?.encryptedFieldsMap;
+      let encryptedFields: Document | undefined =
+        options.encryptedFields ?? encryptedFieldsMap?.[`${db.databaseName}.${name}`];
+
+      if (!encryptedFields && encryptedFieldsMap) {
+        // If the MongoClient was configued with an encryptedFieldsMap,
+        // and no encryptedFields config was available in it or explicitly
+        // passed as an argument, the spec tells us to look one up using
+        // listCollections().
+        const listCollectionsResult = await db
+          .listCollections({ name }, { nameOnly: false })
```

Comment:
It is – I wanted to be explicit here, since we are specifically using `listCollections` to look up a field other than name/type. Happy to adjust this, of course :slightly_smiling_face:

Reply:
I'm not too bothered by it being explicitly called out here, just a question.